core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,396 @@
+    const currentOption = this.options.find(
+      (option) => option.contributionType === contributionType);
+    this.selectedContributionType = currentOption.displayName;
+    this.dropdownShown = false;
+    this.mobileDropdownShown = false;
+  }
+
+  async fetchStats(): Promise<void> {
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      this.username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      response.translation_contribution_stats.map((stat) => {
+        const language = this.languageUtilService.getAudioLanguageDescription(
+          stat.language_code);
+        if (!this.statsData?.translationContribution[language]) {
+          this.statsData.translationContribution[language] = this
+            .createTranslationContributionPageableStats(stat);
+        } else {
+          this.statsData.translationContribution[language].data.push(
+            this.createTranslationContributionStat(stat));
+        }
+      });
+    }
+
+    if (response.translation_review_stats.length > 0) {
+      response.translation_review_stats.map((stat) => {
+        const language = this.languageUtilService.getAudioLanguageDescription(
+          stat.language_code);
+        if (!this.statsData.translationReview[language]) {
+          this.statsData.translationReview[language] = this
+            .createTranslationReviewPageableStats(stat);
+        } else {
+          this.statsData.translationReview[language].data.push(
+            this.createTranslationReviewStat(stat));
+        }
+      });
+    }
+
+    if (response.question_contribution_stats.length > 0) {
+      this.statsData.questionContribution = this
+        .createQuestionContributionPageableStats(
+          response.question_contribution_stats);
+    }
+
+    if (response.question_review_stats.length > 0) {
+      this.statsData.questionReview = this.createQuestionReviewPageableStats(
+        response.question_review_stats);
+    }
+  }
+
+  createTranslationContributionPageableStats(
+      stat: TranslationContributionBackendDict): PageableStats {
+    return {
+      data: [this.createTranslationContributionStat(stat)],
+      currentPageStartIndex: 0
+    };
+  }
+
+  createTranslationReviewPageableStats(
+      stat: TranslationReviewBackendDict): PageableStats {
+    return {
+      data: [this.createTranslationReviewStat(stat)],
+      currentPageStartIndex: 0
+    };
+  }
+
+  createQuestionContributionPageableStats(
+      stats: QuestionContributionBackendDict[]): PageableStats {
+    return {
+      data: stats.map((stat) => {
+        return this.createQuestionContributionStat(stat);
+      }),
+      currentPageStartIndex: 0
+    };
+  }
+
+  createQuestionReviewPageableStats(
+      stat: QuestionReviewBackendDict[]): PageableStats {
+    return {
+      data: stat.map((stat) => {
+        return this.createQuestionReviewStat(stat);
+      }),
+      currentPageStartIndex: 0
+    };
+  }
```

Comment:
Instead of this let's just make `PageableStats` a class instead of an interface with a `create(data)` method. Especially since we always set `currentPageStartIndex: 0`.

Reply:
Done